Assets/PackageStagingArea/eu.netherlands3d.twin-functionalities/Scripts/Functionality.cs

```diff
@@ -17,36 +47,35 @@ public class Functionality : ScriptableObject, ISimpleJsonMapper
 
         [Tooltip("The header above the description")]
         public string Header;
-        [TextArea(5, 10)]
-        public string Description;
-        public ScriptableObject configuration;
 
-        [SerializeField] private bool isEnabled;
+        [TextArea(5, 10)] public string Description;
+        public ScriptableObject configuration;
 
+        public string Id => Data.Id;
+        
         public bool IsEnabled
         {
-            get => isEnabled;
+            get => Data.IsEnabled;
             set
             {
                 var config = configuration as IConfiguration;
                 var targetValue = value;
 
                 //cant enable functionality if configuration is invalid
-                if (isEnabled && config != null && config.Validate().Count > 0){
+                if (Data.IsEnabled && config != null && config.Validate().Count > 0)
+                {
                     Debug.LogWarning($"Can't enable functionality {Title} because configuration is invalid");
                     targetValue = false;
                 }
 
-                var wasEnabled = isEnabled;
-                isEnabled = value;
-                switch (wasEnabled)
+                if (value != Data.IsEnabled) //IsEnabled was changed
```

Comment:
Style puntje: als je deze if omdraait kan er een nesting niveau minder hieronder. Dus:

```
if (value == Data.IsEnabled) return;

Data.IsEnabled = value;
if (Data.IsEnabled)
    OnEnable.Invoke();
else
    OnDisable.Invoke();
```

Dit is een trucje om de complexiteit van je code te verminderen door het aantal onnodige nesting niveaus terug te dringen. Dit komt o.a. uit het idee van early returns uit Object calesthenics (https://medium.com/@davidsen/clean-code-object-calisthenics-f6f4dec07c8b)

Reply:
Het was eerst een mooie switch statement die ik niet begreep, dus dit was al een verbetering, maar ik zal het nog verder versimpelen ;)